Source/Core/DolphinQt2/MainWindow.cpp

```diff
@@ -335,11 +340,17 @@ bool MainWindow::Stop()
                                            "may be lost if you stop the current emulation "
                                            "before it completes. Force stop?") :
                                         tr("Do you want to stop the current emulation?"));
+
+    m_inquiry_in_progress = false;
+
     if (pause)
       Core::SetState(state);
 
     if (confirm != QMessageBox::Yes)
+    {
+      m_exit_requested = false;
```

Comment:
No, I don't think it can, as it cannot tell at that point whether or not the stop was invoked by closing the widget or the window. If you do that in the eventFilter, however you can't differentiate between cancelling a shutdown (Pressing No) and deciding to wait for a safe shutdown.